src/main/kotlin/io/github/nomisrev/service/ProfileService.kt

```diff
@@ -0,0 +1,21 @@
+package io.github.nomisrev.service
+
+import arrow.core.Either
+import io.github.nomisrev.DomainError
+import io.github.nomisrev.repo.UserPersistence
+import io.github.nomisrev.routes.Profile
+
+interface ProfileService {
+  /** Select a Profile by its username */
+  suspend fun getProfile(username: String): Either<DomainError, Profile>
+}
+
+fun profileService(
+  repo: UserPersistence,
+): ProfileService =
+  object : ProfileService {
+
+    override suspend fun getProfile(
+      username: String,
+    ): Either<DomainError, Profile> = repo.selectProfile(username)
```

Comment:
**Nit:** in cases that repo maps 1-on-1 with service I skip the service. So this can be removed following the rule of thumb to "not prematurely optimise".

Reply:
I can see that, there's a constant discussion regarding whether we should keep it consistent or try "not to optimize prematurely".

I do agree that it's just unnecessary noise in this case.

Addresed in [637f167](https://github.com/nomisRev/ktor-arrow-example/pull/186/commits/637f1676c54c35c71d52b6d1ba79d1b17d8408df)